modules/client/src/main/java/org/apache/ignite/internal/client/table/AbstractClientView.java

```diff
@@ -50,6 +63,36 @@ abstract class AbstractClientView<T> implements CriteriaQuerySource<T> {
         this.tbl = tbl;
     }
 
+    /**
+     * Get index mapping from result set to schema.
+     *
+     * @param columns Columns to map.
+     * @param metadata Metadata for query results.
+     * @param startInclusive The first index to cover.
+     * @param endExclusive Index immediately past the last index to cover.
+     * @return Index mapping.
+     */
+    protected static List<Integer> indexMapping(
+            ClientColumn[] columns,
+            int startInclusive,
+            int endExclusive,
+            ResultSetMetadata metadata
+    ) {
+        return Arrays.stream(columns, startInclusive, endExclusive)
+                .map(ClientColumn::name)
+                .map(IgniteNameUtils::quoteIfNeeded)
+                .map((columnName) -> {
+                    int rowIdx = metadata.indexOf(columnName);
+
+                    if (rowIdx == -1) {
+                        throw new IgniteException(Common.INTERNAL_ERR, "Missing required column in query results: " + columnName);
+                    }
+
+                    return rowIdx;
+                })
+                .collect(toList());
```

Comment:
let's avoid boxing results here and for all paths of usage

Reply:
Fixed